server/game/cards/events/07/scalingthewall.js

```diff
@@ -0,0 +1,50 @@
+const DrawCard = require('../../../drawcard.js');
+
+class ScalingTheWall extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                afterChallenge: (event, challenge) => (
+                    challenge.winner === this.controller &&
+                    challenge.attackingPlayer === this.controller &&
+                    this.hasAttackingWildling()
+                )
+            },
+            handler: () => {
+                this.game.promptForSelect(this.controller, {
```

Comment:
Instead of a manual prompt, passing a target property into the reaction is less work and makes sure the reaction doesn't prompt when there are no legal targets. Something like this (it's explained in more depth in the card implementation docs):

            target: {
                activePromptTitle: 'Select location',
                cardCondition: card => card.getType() === 'location' && !card.isLimited();
            }